docs/proposals/platform-extension.md

```diff
@@ -0,0 +1,105 @@
+
+### Current Addon Mechanism
+#### Main Concept Points
+
+* Default Addons (core addons)
+  * Always installed on each user cluster, like
+    * Canal
+    * Dashboard
+    * kube-proxy
+
+
+* Accessible addons
+  * Addons that the user is allowed to interact with.
+    * Node Exporter
+    * KubeVirt / CDI (implementation dropped because of addon complexity)
+    * KubeFlow
+
+* Custom Addons
+  * Addons that are installed by default together with core addons. Requires to ship a custom docker image
+
+#### Summary
+##### Pros
+* Good enough for simple and core deployments that users don’t have a chance to interact with like:
+  * kube-proxy
+  * CSI drivers
+  * OpenVPN
+
+##### Cons
+* Difficult to extend
+  * Accessible addons shipped as a docker image together with core addons
+* Kubermatic upgrade enforce to rebuild the addon image
+* Difficult to deploy more complex K8s addons/apps:
+  * Since the plugin contains multiple CRDs and their instantiations at the same time, it takes several attempts in reconciliation loop until it is successfully applied. That results into several warnings in the cluster event log in the KKP UI.
+  * Since KubeCarrier requires cert-manager to run, cert-manager has to be installed prior to KubeCarrier. If they are both installed at once as part of the same plugin, it takes long time (several minutes) until everything converges and KubeCarrier is ready. Sometimes, KubeCarrier does not even start at all.
+* No control over dependencies, the requirement:
+  * A KKP addon may be dependent on a different addon (e.g. addon A creates a CRD / resource that addon B instantiates / consumes). An example of this is the KubeCarrier addon, which is dependent on the cert-manger addon, that needs to be deployed in before.
+
+#### Action Items
+* Remove accessible addons from KKP and replace them with Platform Extensions
+* Drop idea of shipping custom image addons
+
+
+
+## Implementation
+
+### Platform Extensions
+
+#### Main Concept Points
+
+* Extensions provided by KKP devs
+  * Extensions created as helm charts.
+  * All extensions in the same repo kubermatic/extensions
+    * The repo tags match the k8s version, for instance v1.19, v1.20
+  * Extensions pushed to quay.io/kubermatic-extensions/&lt;extension>:&lt;tag>
+* Extensions provided by customers
+  * Can point to any OCI registry
+  * Customers should keep our tagging pattern
+* Extensions are **not** tied to the KKP lifecycle
+* Extensions are tied to the K8S or their own version lifecycle
+* Extensions are editable
+
+#### Implementation
```

Comment:
The individual implementations then need own documents, like `platform-extension-foo.md` and  `platform-extension-bar.md`. So we have more room for the designs.